### ISSUE TITLE
Async VariableStorage issues

### ISSUE THREAD

**Issue by kenerwin88** [mnbroatch/yarn-bound#5]
Hi,

First, I love your yarn-bound package so much,  it’s made it so I can finally use yarn in a JS game I’m making.  Secondly, I know how much work it is to maintain an open source tool, and I would be more than happy to pay you to fix the issue as I know it’s an edge case but I really need it.  (You can reach me at ken@devopslibrary.com if you’re up for it!).

The problem is that I’m trying to store the variables in a database (vs. the map in the example).  I’ve implemented all of it, but the one problem I have is that my “get” function has to be async, as it’s querying the database.  There’s no way around it that I can think of that isn’t super hacky.

Within yarn-bound, within bondage.js, the conditional check (like say you’re doing “show this choice if X is set”, otherwise don’t show it), doesn’t evaluate correctly, as it receives a promise, not a value if the “get” of the variable storage is async.

In other words, would it be possible to make yarn-bound support async variable storage?  If you don’t have time I completely understand.

**Comment by mnbroatch** [comment]
Unfortunately, it seems at first glance that doing so would cause the entire YarnBound app to be necessarily async. That is, all the internals, and more crucially the base API `runner.advance` would have to start returning promises. This would be too detrimental to the default behavior. I would support a fork like yarnbound-async or something, but I'm not interested in building that feature or incorporating it into the main branch. I can certainly give some guidance if you wanted to tackle that.

But I think the best thing to do would be to improve your hacky solution, incorporating it in the application layer above this library. I'm also happy to discuss design challenges along that route. 

My first idea would be that your "advance" button, say, should fetch necessary data, then when it's fetched and your variable store has been updated, only then run `runner.advance()`. There are several ways you could indicate in the yarn file which variables will be needed (custom node properties, hashtags, a custom prefetch command), or your app could be more intelligent about it.

**Comment by kenerwin88** [comment]
Thank you for the response!! I think you’re 100% right, it’s just not worth it.  I think what I can do is similar to what you’re saying, and actually fetch everything from the DB that I know I’ll need on map load, before the player has even initiated a convo.  At that point, it’s stored in a map, which I can access synchronously.  

In other words, thank you 😀
